catalog/resource_metastore_test.go

```diff
@@ -25,8 +26,15 @@ func TestCreateMetastore(t *testing.T) {
 				},
 			},
 			{
-				Method:   "GET",
+				Method:   "PATCH",
```

Comment:
It seems like this is introduced because in the old updateFunctionFactory, we checked if there were any fields that needed to be updated. Now, we always call update, even when it is a no-op. I think this should be safe (it just adds one extra round trip that doesn't change the underlying resource).

Reply:
yes - we could try to be more precise and check for presence of fields that are only updatable, but it feels like too much extra logic to maintain

Reply:
I think something should be possible with reflection (i.e. check whether there is any field in the update struct that is not the zero value for its type), but I would rather scope that out of this PR.